api/bootstrap/kubeadm/v1beta2/kubeadm_types.go

```diff
@@ -199,6 +216,13 @@ type ClusterConfiguration struct {
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=36500
 	CACertificateValidityPeriodDays int32 `json:"caCertificateValidityPeriodDays,omitempty"`
+
+	// encryptionAlgorithm holds the type of asymmetric encryption algorithm used for keys and certificates.
+	// Can be one of "RSA-2048", "RSA-3072", "RSA-4096", "ECDSA-P256" or "ECDSA-P384".
+	// If not specified, Cluster API will use RSA-2048 as default.
+	// This field is only supported with Kubernetes v1.31 or above.
+	// +optional
+	EncryptionAlgorithm EncryptionAlgorithmType `json:"encryptionAlgorithm,omitempty"`
```

Comment:
Let's set this field to RSA-4096 in test/e2e/data/infrastructure-docker/main/clusterclass-quick-start-runtimesdk.yaml KubeadmControlPlaneTemplate for some free e2e test coverage